pom.xml

```diff
@@ -2786,7 +2786,7 @@
         <artifactId>maven-checkstyle-plugin</artifactId>
         <version>3.1.0</version>
         <configuration>
-          <failOnViolation>false</failOnViolation>
+          <failOnViolation>true</failOnViolation>
```

Comment:
Alternatively, we can just completely switch `linter-java` to `sbt-checkstyle` completely so that we don't suffer from the differences anymore.

I have seen some differences so far, e.g. https://github.com/apache/spark/pull/26818 and https://github.com/apache/spark/pull/21801